### PR TITLE
Documentation regarding to NGSIv2 registration (except missing functionality)

### PR DESCRIPTION
**Pull request by fgalan**
The changes in this PR assume that DELETE /v2/registration/{id} has been implemented (except for the diagraman and associated documentation in development manual).

### CODE REVIEW

File: doc/manuals/admin/database_model.md [hunk 1]
```diff
@@ -193,12 +193,14 @@ Fields:
     we ensure that registration IDs are unique and that queries by
     registration IDs will be very fast (as there is an automatic default
     index in \_id).
+-   **format**: the format to use to send forwarded request. Possible value
```
Comment (kzangeli) [inline doc/manuals/admin/database_model.md]: The only accepted value for now is ...

Comment (kzangeli) [reply doc/manuals/admin/database_model.md]: to send forwarded request*s*

File: doc/manuals/admin/database_model.md [hunk 2]
```diff
@@ -210,7 +212,9 @@ Fields:
     following information:
     -   **entities**: an array containing a list of
         entities (mandatory). The JSON for each entity contains **id**,
-        **type** and **isPattern**.
+        **type**, **isPattern** and "isTypePattern". Note that, due to legacy
```
Comment (kzangeli) [inline doc/manuals/admin/database_model.md]: Should `and "isTypePattern"` be bold, like `**type**, **isPattern**`   ?

File: doc/manuals/admin/database_model.md [hunk 3]
```diff
@@ -301,7 +306,7 @@ Fields:
     the subscription.   
 -   **format**: the format to use to send notification, possible values are **JSON**
```
Comment (kzangeli) [inline doc/manuals/admin/database_model.md]: What about "legacy" ...
That's also valid, right?

File: doc/manuals/devel/mongoBackend.md [hunk 4]
```diff
@@ -580,6 +582,65 @@ _MB-21: mongoUnsubscribeContextAvailability_
...
+
+<a name="flow-mb-24"></a>
+![mongoRegistrationsGet](images/Flow-MB-24.png)
+
+_MB-24: mongoRegistrationsGet_
+
+* `mongoRegistrationsGet()` is invoked from a service routine (step 1).
+* Depending on `-reqMutexPolicy`, the request semaphore may be taken (read mode) (step 2). See [this document for details](semaphores.md#mongo-request-semaphore). 
+* The subscription is retrieved from the database using `collectionRangedQuery()` in the `connectionOperations` module (steps 3 and 4).
+* For each subscription to return, several `set*()` functions are used in order to fill the `Registration` objects.
```
Comment (kzangeli) [inline doc/manuals/devel/mongoBackend.md]: For each subscription to return ...   "s/subscription/registration/" ?

File: doc/manuals/admin/database_model.md (hunk 1 above)
Comment (fgalan) [reply doc/manuals/admin/database_model.md]: Fixed by 1c7c242 (I guess)

File: doc/manuals/admin/database_model.md (hunk 2 above)
Comment (fgalan) [reply doc/manuals/admin/database_model.md]: Right. I'll change.

File: doc/manuals/devel/mongoBackend.md (hunk 4 above)
Comment (fgalan) [reply doc/manuals/devel/mongoBackend.md]: I'll fix

File: doc/manuals/user/v1_v2_coexistence.md [hunk 5]
```diff
@@ -128,3 +129,22 @@ However, the following considerations have to be taken into account:
...
+
+In general, there is no problem creating registrations using NGSIv1 (in particular, `POST /v1/registry/registerContext`) and then retrieving them using NGSIv2 (in particular, `GET /v2/registrations` or `GET /v2/registrations/<id>`).
+
+Note that NGSIv1 considers the concept of "context registration". A registration is composed of several context
+registrations, each one being composed of a set of entities and attributes. NGSIv2 proposes a much simpler approach,
+without context registration as intermediate element, i.e. a registration is associated to a set of entities and
+attributes directly.
+
+In the case of retrieving a registration created using NGSIv1, and that has more than one context registration, with
+`GET /v2/registrations` or `GET /v2/registrations/<id>` only the first one is consideren. In other words,
```
Comment (kzangeli) [inline doc/manuals/user/v1_v2_coexistence.md]: Typo: consideren => considered

File: src/lib/mongoBackend/mongoRegistrationGet.cpp [hunk 6]
```diff
@@ -84,11 +84,20 @@ static void setProvider(ngsiv2::Registration* regP, const mongo::BSONObj& r)
...
-  // FIXME P4: for the moment supportedForwardingMode and legacyForwardingMode are hardwired (i.e. DB is not taken
-  // into account for them)
+  // FIXME #3106: for the moment supportedForwardingMode is hardwired (i.e. DB is not taken
+  // into account for them).
   //
   regP->provider.supportedForwardingMode = ngsiv2::ForwardAll;
-  regP->provider.legacyForwardingMode = true;
+
+  std::string format = r.hasField(REG_FORMAT)? getStringFieldF(r, REG_FORMAT) : "JSON";
+  if (format == "JSON")
```
Comment (kzangeli) [inline src/lib/mongoBackend/mongoRegistrationGet.cpp]: Same doubt here. What about "legacy".
I am probably wrong ...

File: doc/manuals/admin/database_model.md (hunk 3 above)
Comment (fgalan) [reply doc/manuals/admin/database_model.md]: No...  `"attrsFormat": "legacy"` uses "JSON", not "legacy". 

Why? I'm not fully sure, but I think this comes from the XML times, where `format` could be "XML" or "JSON". The "XML" alternative disappeared in Orion 1.0.0, leaving "JSON". So, new format used "keyValues", etc. leaving old "JSON" for NGSIv1 format (i.e. legacy notifications).

File: src/lib/mongoBackend/mongoRegistrationGet.cpp (hunk 6 above)
Comment (fgalan) [reply src/lib/mongoBackend/mongoRegistrationGet.cpp]: Already answered in the other comment.

File: doc/manuals/user/v1_v2_coexistence.md (hunk 5 above)
Comment (fgalan) [reply doc/manuals/user/v1_v2_coexistence.md]: Fixed in d4bc65f

File: doc/manuals/devel/mongoBackend.md (hunk 4 above)
Comment (fgalan) [reply doc/manuals/devel/mongoBackend.md]: Fixed in d4bc65f

File: doc/manuals/admin/database_model.md (hunk 2 above)
Comment (fgalan) [reply doc/manuals/admin/database_model.md]: Fixed in d4bc65f

File: src/lib/mongoBackend/mongoRegistrationGet.cpp (hunk 6 above)
Comment (kzangeli) [reply src/lib/mongoBackend/mongoRegistrationGet.cpp]: NTC

File: doc/manuals/admin/database_model.md (hunk 3 above)
Comment (kzangeli) [reply doc/manuals/admin/database_model.md]: NTC

**Comment by fgalan** [comment]
Passing the ball to @jmcanterafonseca for final LGTM

**Pull request merged**